### ISSUE TITLE
can't remember the last window's size

### ISSUE THREAD

**Issue by 2alin** [u8sand/Baka-MPlayer#185]
I'm on Ubuntu 15.04, and I just tried this player since yesterday (I was using SMPlayer +mpv) and I found it a pretty good option as default video player, but there was only one thing that really annoyed me. The player can't remember the window's size or if it was maximized in the last session. So, everytime I open a video (even the same one) it maps the window to an exactly fixed small size (1/4 of my screen or so). 
Furthermore, if I have the player maximized, go to fullscreen and then I get out from fullscreen; the player doesn't send me to the maximized mode again, instead it send me to small restored window.

**Comment by u8sand** [comment]
We used to remember the screen size but removed it in favor of the automatic FitWindow feature. I would recommend, if the videos you usually watch are much smaller than your screen--setting the autofit window percentage to be 200% or whatever tends to work. Please note that when your video is open you can automatically scale it proportionally to your screen with Alt+1/Alt+2/Alt+3/... (bigger).

I often find it more ideal to fit videos big and use the dim feature, my desktop is nicer than the black border of Baka MPlayer's maximized window. Have you used the FitWindow feature and do you still prefer maximizing and/or manual re-sizing?

**Comment by 2alin** [comment]
Yup, I've being using FitWindow at 200% and Ctrl+D; some videos with 100% are enough, but some doesn't and I find it troublesome to change the percentage, even with keyboard shortcuts everytime I open a video. Your player is better than SMPlayer manipulating x265 videos and the audio in your player is better too, so I'm going to stick with it for a while.

But I still prefer the normal "remembering screen mode/size" feature, not all the videos that I have are 1080p or 720p, and for smaller ones 200% is not even enough to watch them in my monitor. Now, I think that it could be great to automatically enable "remembering screen" when the FitWindow feature is disabled. 646x400 is too small for screens with 1080p resolution. You could have both feautures at hand.

Anyway, thank you so much. Your player is awesome, and even if it's not perfect, the way it handles x265 videos, the player's look, the features... makes me love it. Thank you!

**Comment by duyfken** [comment]
I have to agree it would be great to either have it remember "when the FitWindow feature is disabled" as 2alin mentioned or at least have some option to remember the previous window size. I previously used this on MPC-HC and bomi to set window size to 1280x720 on my 2560x1080 ultrawide (to take up half the screen and leave space for doing stuff as well) and having to resize the window every time is one of the frustrations preventing me moving from (the currently unmaintained) bomi to Baka atm.

**Comment by u8sand** [comment]
Should I also save position on screen, or just center it with your old size?

**Comment by 2alin** [comment]
I think for a better wide range of personalization, it would be better to remember the size as well. Some people would love to put a file explorer at the left of the screen and the player at the right, while browsing files in multiple directories, and remembering size and position will make the video browsing smooth as heck.
And thank you for reading our opinions, u8sand.

**Comment by ConundrumAdInfinitum** [comment]
Using "View > FitWindow > size" works, but I am a bit confused why "View > AspectRatio > size" isn't constraining the window to the exact ratio you select. I feel it would make sense to snap all the borders to the selected Aspect Ratio. Also, couldn't the FitWindow and AspectRatio items be condensed into one menu? Logically you would be "Fitting Window" to a percentage or available aspect ratio. It seems like excess to have them split apart into two menus.

**Comment by goetzc** [comment]
Also remembering if the window was maximized or not would be useful.
